lib/Doctrine/Common/Cache/ChainCache.php

```diff
@@ -52,7 +52,34 @@ public function setNamespace($namespace)
             $cacheProvider->setNamespace($namespace);
         }
     }
+    
+    /**
+     * {@inheritDoc}
+     */
+    protected function doFetchMultiple(array $keys)
```

Comment:
Can't you simply defer to `fetchMultiple` to sub-providers, if available and the interface supports that?